<?xml version="1.0"?>
<launch>
    <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find cvg_sim_gazebo)/worlds/markerworld.world"/>
    </include>
    <node pkg="cvqiao" type="genBackRef.py" name="genLeaderRef" output="screen"/>

    <node pkg="cvqiao" type="genFrontRef.py" name="genFollowerRef" output="screen"/>
    <node pkg="cvqiao" type="controlAllUAV.py" name="controlUAV" output="screen" launch-prefix="xterm -geometry 150x20+0+0 -e" />
    <group ns="drone2">
        <!-- Spawn simulated quadrotor uav -->
        <include file="$(find cvg_sim_gazebo)/launch/spawn_quadrotor.launch" >
            <arg name="model" value="$(find cvg_sim_gazebo)/urdf/quadrotor_with_sensors.urdf.xacro"/>
            <arg name="model_name" value="drone2"/>
            <arg name="x" value="2.5"/>
            <arg name="y" value="-0.5"/>
            <arg name="z" value="0"/>
            <arg name="tf_prefix" value="/drone2"/>
            <arg name="namespace" value="/drone2"/>
            <arg name="marker_NO" value="3"/>
        </include>
	<include file="$(find cvqiao)/launch/test1sim2drone23.launch">
        <arg name="px_init" value="2.5"/>
        <arg name="py_init" value="-0.5"/>
        <arg name="pz_init" value="0"/>
    </include>
    <node pkg="rosbag" type="record" name="myrosbag"  output="screen" args="str_log /gazebo/model_states cmd_vel err cmd_world mixIMUMarker mixMarker genRef from_IMU tello/odom mixMarkerSTD marker_10  cam_from_marker_10 marker_11  cam_from_marker_11 -o drone2"/>
    </group>
    <group ns="drone3">
        <!-- Spawn simulated quadrotor uav -->
        <include file="$(find cvg_sim_gazebo)/launch/spawn_quadrotor.launch" >
            <arg name="model" value="$(find cvg_sim_gazebo)/urdf/quadrotor_with_sensors.urdf.xacro"/>
            <arg name="model_name" value="drone3"/>
            <arg name="x" value="2.5"/>
            <arg name="y" value="0.5"/>
            <arg name="z" value="0"/>
            <arg name="tf_prefix" value="/drone3"/>
            <arg name="namespace" value="/drone3"/>
            <arg name="marker_NO" value="4"/>
        </include>
	<include file="$(find cvqiao)/launch/test1sim2drone23.launch">
        <arg name="px_init" value="2.5"/>
        <arg name="py_init" value="0.5"/>
        <arg name="pz_init" value="0"/>
    </include>
    <node pkg="rosbag" type="record" name="myrosbag"  output="screen" args="str_log /gazebo/model_states cmd_vel err cmd_world mixIMUMarker mixMarker genRef from_IMU tello/odom mixMarkerSTD marker_10  cam_from_marker_10 marker_11  cam_from_marker_11 -o drone3"/>
    </group>



    <group ns="drone1">
        <include file="$(find cvqiao)/launch/test1sim.launch">
            <arg name="px_init" value="4"/>
            <arg name="py_init" value="0"/>
            <arg name="pz_init" value="0"/>
        </include>
        <include file="$(find cvg_sim_gazebo)/launch/spawn_quadrotor.launch" >
            <arg name="model" value="$(find cvg_sim_gazebo)/urdf/quadrotor_with_sensors.urdf.xacro"/>
            <arg name="model_name" value="drone1"/>
            <arg name="x" value="4"/>
            <arg name="y" value="0"/>
            <arg name="z" value="0"/>
            <arg name="tf_prefix" value="/drone1"/>
            <arg name="namespace" value="/drone1"/>
        </include>
    <node pkg="rosbag" type="record" name="myrosbag"  output="screen" args="str_log /gazebo/model_states cmd_vel err cmd_world mixIMUMarker mixMarker genRef from_IMU tello/odom mixMarkerSTD marker_0 marker_1 marker_2 marker_3 marker_4 marker_5 marker_6 cam_from_marker_0 cam_from_marker_1 cam_from_marker_2 cam_from_marker_3 cam_from_marker_4 cam_from_marker_5 cam_from_marker_6 -o drone1"/>

    </group>

</launch>
